app/javascript/courses/students/components/CoursesStudents__CoachNoteShow.re

```diff
@@ -0,0 +1,137 @@
+    |> GraphqlQuery.sendQuery(AuthenticityToken.fromHead())
+    |> Js.Promise.then_(response => {
+         if (response##archiveCoachNote##success) {
+           removeNoteCB(id);
+         } else {
+           setArchiving(_ => false);
+         };
+         Js.Promise.resolve();
+       })
+    |> ignore;
+  } else {
+    ();
+  };
+};
+
+let deleteIcon =
+    (note, removeCoachNote, removeNoteCB, setArchiving, archiving) => {
+  <button
+    className="w-10 text-sm text-gray-700 hover:text-gray-900 cursor-pointer flex items-center justify-center rounded hover:bg-gray-200 focus:outline-none "
+    disabled=archiving
+    title={"Delete " ++ (note |> CoachNote.id)}
+    onClick={removeCoachNote(
+      note |> CoachNote.id,
+      removeNoteCB,
+      setArchiving,
+    )}>
+    <FaIcon
+      classes={archiving ? "fas fa-spinner fa-spin" : "fas fa-trash-alt"}
+    />
+  </button>;
+};
+
+[@react.component]
+let make = (~note, ~userId, ~removeNoteCB) => {
+  let (archiving, setArchiving) = React.useState(() => false);
+  <div
+    className="mt-4"
+    key={note |> CoachNote.id}
+    ariaLabel={"Note " ++ (note |> CoachNote.id)}>
+    <div className="flex justify-between">
+      <div className="flex">
+        {switch (note |> CoachNote.author) {
+         | Some(coach) =>
+           switch (coach |> Coach.avatarUrl) {
+           | Some(avatarUrl) =>
+             <img
+               className="w-8 h-8 md:w-10 md:h-10 text-xs border border-gray-400 rounded-full overflow-hidden flex-shrink-0 mt-1 md:mt-0 mr-2 md:mr-3 object-cover"
+               src=avatarUrl
+             />
+           | None =>
+             <Avatar
+               name={coach |> Coach.name}
+               className="w-8 h-8 md:w-10 md:h-10 text-xs border border-gray-400 rounded-full overflow-hidden flex-shrink-0 mt-1 md:mt-0 mr-2 md:mr-3 object-cover"
+             />
+           }
+
+         | None =>
+           <Avatar
+             name="?"
+             className="w-8 h-8 md:w-10 md:h-10 text-xs border rounded-full overflow-hidden flex-shrink-0 mt-1 md:mt-0 mr-2 md:mr-3 object-cover"
+           />
+         }}
+        <div>
+          <p className="text-sm font-semibold inline-block leading-snug">
+            {(
+               switch (note |> CoachNote.author) {
+               | Some(coach) => coach |> Coach.name
+               | None => "Deleted Coach"
+               }
+             )
+             |> str}
+          </p>
+          <p
+            className="text-gray-600 font-semibold text-xs mt-px leading-snug">
+            {(
+               switch (note |> CoachNote.author) {
+               | Some(coach) => coach |> Coach.title
+               | None => "Unknown"
+               }
+             )
+             |> str}
+          </p>
+        </div>
+      </div>
+      {let showDeleteIcon =
+         switch (note |> CoachNote.author) {
+         | None => false
+         | Some(coach) => Coach.id(coach) == userId
```

Comment:
Comparing `Coach.id` to `userId` is a red flag! See comment in `CoursesStudents__Coach.re`.

Reply:
Fixed in efe77441